server/src/main/java/org/opensearch/gateway/TransportNodesGatewayStartedShardHelper.java

```diff
@@ -0,0 +1,189 @@
+            );
+            return exception;
+        }
+        return null;
+    }
+
+    /**
+     * Helper function for getting the string representation of the NodeGatewayStartedShards object
+     *
+     * @param allocationId
+     * @param primary
+     * @param storeException
+     * @param replicationCheckpoint
+     * @return String
+     */
+    public static String NodeGatewayStartedShardsToString(
+        String allocationId,
+        boolean primary,
+        Exception storeException,
+        ReplicationCheckpoint replicationCheckpoint
+    ) {
+        StringBuilder buf = new StringBuilder();
+        buf.append("NodeGatewayStartedShards[").append("allocationId=").append(allocationId).append(",primary=").append(primary);
+        if (storeException != null) {
+            buf.append(",storeException=").append(storeException);
+        }
+        if (replicationCheckpoint != null) {
+            buf.append(",ReplicationCheckpoint=").append(replicationCheckpoint.toString());
+        }
+        buf.append("]");
+        return buf.toString();
+    }
+
+    /**
+     * Helper function for computing the hashcode of the NodeGatewayStartedShards object
+     *
+     * @param allocationId
+     * @param primary
+     * @param storeException
+     * @param replicationCheckpoint
+     * @return int
+     */
+    public static int NodeGatewayStartedShardsHashCode(
+        String allocationId,
+        boolean primary,
+        Exception storeException,
+        ReplicationCheckpoint replicationCheckpoint
+    ) {
+        int result = (allocationId != null ? allocationId.hashCode() : 0);
+        result = 31 * result + (primary ? 1 : 0);
+        result = 31 * result + (storeException != null ? storeException.hashCode() : 0);
+        result = 31 * result + (replicationCheckpoint != null ? replicationCheckpoint.hashCode() : 0);
+        return result;
+    }
+
+    /**
+     * Helper function for NodeGatewayStartedShardsWriteTo method
+     *
+     * @param out
+     * @param allocationId
+     * @param primary
+     * @param storeException
+     * @param replicationCheckpoint
+     * @throws IOException
+     */
+    public static void NodeGatewayStartedShardsWriteTo(
+        StreamOutput out,
+        String allocationId,
+        boolean primary,
+        Exception storeException,
+        ReplicationCheckpoint replicationCheckpoint
+    ) throws IOException {
+        out.writeOptionalString(allocationId);
+        out.writeBoolean(primary);
+        if (storeException != null) {
+            out.writeBoolean(true);
+            out.writeException(storeException);
+        } else {
+            out.writeBoolean(false);
+        }
+        if (out.getVersion().onOrAfter(Version.V_2_3_0)) {
+            if (replicationCheckpoint != null) {
+                out.writeBoolean(true);
+                replicationCheckpoint.writeTo(out);
+            } else {
+                out.writeBoolean(false);
+            }
+        }
+    }
+}
```

Comment:
I don't think so you need a Helper class here. All of these can be modelled as properties of a class

Reply:
Used composition pattern in the next commit.

Reply:
is this addressed, I still see the helper class being present.